compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/AnnotationDesugar.java

```diff
@@ -191,6 +199,84 @@ private void defineClassAnnotations(BLangPackage pkgNode, SymbolEnv env, BLangFu
         }
     }
 
+    private boolean isServiceDeclaration(BLangClassDefinition serviceClass) {
+        return Objects.nonNull(serviceClass.getFlags())
+                && serviceClass.getFlags().contains(Flag.SERVICE)
+                && (serviceClass.isServiceDecl || serviceClass.getFlags().contains(Flag.OBJECT_CTOR));
+    }
+
+    private void addIntrospectionInfoAnnotation(BLangClassDefinition serviceClass) {
+        Location position = serviceClass.pos;
+
+        // Create Annotation Attachment.
+        BLangAnnotationAttachment annoAttachment = (BLangAnnotationAttachment) TreeBuilder.createAnnotAttachmentNode();
+        final SymbolEnv pkgEnv = symTable.pkgEnvMap.get(serviceClass.symbol.getEnclosingSymbol());
+        BSymbol annSymbol = symResolver.lookupSymbolInAnnotationSpace(
+                pkgEnv, names.fromString(SERVICE_INTROSPECTION_INFO_ANN));
```

Comment:
Given that we know where this annotation is defined, wondering if we should directly do

```suggestion
        symResolver.lookupSymbolInAnnotationSpace(symTable.pkgEnvMap.get(symTable.rootPkgSymbol), 
                                                  names.fromString(SERVICE_INTROSPECTION_INFO_ANN));
```

Reply:
@pubudu91 thoughts please?

Reply:
Yeah, I guess we can do that